src/java/org/apache/cassandra/index/sai/StorageAttachedIndexGroup.java

```diff
@@ -249,9 +249,9 @@ public void handleNotification(INotification notification, Object sender)
         {
             SSTableAddedNotification notice = (SSTableAddedNotification) notification;
 
-            // Avoid validation for index files just written following Memtable flush.
-            IndexValidation validate = notice.memtable().isPresent() ? IndexValidation.NONE : IndexValidation.CHECKSUM;
-            onSSTableChanged(Collections.emptySet(), notice.added, indexes, validate);
+            // Avoid validation for index files just written following Memtable flush. Otherwise, the new SSTables have
+            // come either from import, streaming, or a standalone tool, where they have also already been validated.
+            onSSTableChanged(Collections.emptySet(), notice.added, indexes, IndexValidation.NONE);
```

Comment:
With this change, I think we can no longer get into the trouble we might find ourselves in w/ https://github.com/apache/cassandra/pull/2460#discussion_r1253762041, and the comment explains why we're not validating at all.

The question is whether I now need to simplify the downstream code. Given we're not validating, some of the downstream codepaths are now never hit, like the checksumming path. Once we get here, index components should have been built and such, so the only thing we're doing is updating the views and potentially marking the index non-queryable, which I think is okay. (i.e. Marking the index non-queryable is the only thing that we can really do, but what I need to confirm is whether we're being strict enough around that.)

Thoughts?

Reply:
This is close to the change that I added in my original patch for CASSANDRA-18653. At this point, we should already know that the indexes are built and valid. If they aren't then we aren't in a position to build them anyway.

Reply:
SIM handles the notification, but it will no longer trigger builds for SAI, given SAI will now return true from `Index#isSSTableAttached()`.